src/plugins/auto-moderation-plugin.ts

```diff
@@ -0,0 +1,161 @@
+import fs from 'node:fs'
+
+import type { PluginContext, PluginInterface } from '../common/plugins.js'
+
+import { EventType, InstanceType, PunishmentType, Severity, ChannelType } from 'src/common/application-event.js'
+import { antiSpamString } from 'src/util/shared-util.js'
+
+enum HeatType {
+  SPAM = 'spam',
+  PROFANITY = 'profanity'
+}
+
+type HeatScore = number
+
+type HeatsTable = Record<HeatType, HeatScore>
+type WarnsTable = Record<HeatType, boolean>
```

Comment:
Use Map instead of native object since you are using `delete` keyword to delete entries dynamically.
```suggestion
type HeatsTable = Map<HeatType, HeatScore>
type WarnsTable = Map<HeatType, boolean>
```